app/model/entity/Group.php

```diff
@@ -413,12 +415,43 @@ public function getAssignments() {
     });
   }
 
+  /**
+   * @ORM\OneToMany(targetEntity="ShadowAssignment", mappedBy="group")
+   */
+  protected $shadowAssignments;
+
+  /**
+   * Map collection of shadow assignments to an array of its ID's
+   * @return string[]
+   */
+  public function getShadowAssignmentsIds(): array {
+    return $this->shadowAssignments->map(function (ShadowAssignment $a) {
+      return $a->getId();
+    })->getValues();
+  }
+
+  /**
+   * @return Collection
+   */
+  public function getShadowAssignments() {
+    return $this->shadowAssignments->filter(function (ShadowAssignment $pointsAssignment) {
+      return $pointsAssignment->getDeletedAt() === null;
+    });
+  }
+
   public function getMaxPoints(): int {
-    return array_reduce(
+    $points = array_reduce(
       $this->getAssignments()->getValues(),
       function ($carry, Assignment $assignment) { return $carry + $assignment->getGroupPoints(); },
       0
     );
+    $points = array_reduce(
```

Comment:
I would probably create two variables and add them up in the return statement here.

Reply:
Will do